packages/content/src/attributes/types/birth/BirthDate.ts

```diff
@@ -37,6 +39,33 @@ export class BirthDate extends AbstractComplexValue implements IBirthDate {
     @validate()
     public year: BirthYear;
 
+    protected static override postFrom<T extends Serializable>(value: T): T {
+        if (!(value instanceof BirthDate)) {
+            throw new ConsumptionError("An unexpected error has occured with a BirthDate.");
+        }
+
+        if ((value.month.value === 2 && value.day.value === 30) || ([2, 4, 6, 9, 11].includes(value.month.value) && value.day.value === 31)) {
+            throw new ValidationError(
+                BirthDate.name,
+                nameof<BirthDate>((x) => x.day),
+                "The BirthDate is not a valid date. The chosen day does not exist in the chosen month."
+            );
+        }
+        if (value.month.value === 2 && !value.isInLeapYear() && value.day.value === 29) {
+            throw new ValidationError(
+                BirthDate.name,
+                nameof<BirthDate>((x) => x.year),
+                "The BirthDate is not a valid date. The 29 February only exists in leap years."
+            );
+        }
```

Comment:
we have a great library (luxon) that handles everything date related for us. This library is also able to create dates from day month and year and validate it afterwards. Call me if you need guidance on how to use it.

Reply:
I now use the isInLeapYear function from Luxon instead of implementing my own. Was that your main concern or should I ideally express the IF queries with Luxon as well?

Reply:
Actually you can just use 

```ts
const isValid = DateTime.fromObject({ day: value.day.value, month: value.month.value, year: value.year.value }).isValid;
```

to check if the date is valid, I don't see a reason to give the user this fine-granular feedback about why this date isn't correct.